ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/plugin/PluginInputContextManager.java

```diff
@@ -53,16 +59,32 @@ public IModel getPluginModel() {
 		return findPluginModel();
 	}
 
+
 	@Override
-	protected void structureChanged(IFile file, boolean added) {
-		// If a plugin.xml file has been added to the project the editor should update
-		if (added && ICoreConstants.PLUGIN_FILENAME_DESCRIPTOR.equalsIgnoreCase(file.getName())) {
-			IProject project = getCommonProject();
-			if (project == null || project.equals(file.getProject())) {
-				monitorFile(file);
-			}
+	public void resourceChanged(IResourceChangeEvent event) {
```

Comment:
is it really desired to replace the pld code?

Reply:
The creation of a new product using the product configuration file creates the plugin.xml file.Due to this the "resourceChanged" api of InputContextManager gets called, but however the monitoredFiles list does not contain the "plugin.xml".Due to this the "structureChanged" api of PluginInputContextManager will never be called. Hence it is redundant and can be removed.
The "structureChanged" api of PluginInputContextManager was checking if plugin.xml was added to the project and if so was adding the plugin.xml to the monitoredFiles list.
This operation is now done in the "resourceChanged" api of the PluginInputContextManager so that the "structureChanged" api of the InputContextManager gets called and the plug-in editor content is updated with the new product extension.